backend/src/ml_space_lambda/data_access_objects/project.py

```diff
@@ -154,15 +159,16 @@ def get_all(
             for i in range(len(project_names)):
                 expressions.append(f":p{i}")
 
-            expression_attribute_values: Dict[str, Any] = {":suspended": False}
+            filter_expressions.append(f"#name IN ({', '.join(expressions)})")
+
             for index, expression in enumerate(expressions):
                 expression_attribute_values[expression] = project_names[index]
 
-            json_response = self._scan(
-                filter_expression=f"suspended = :suspended AND #name IN ({', '.join(expressions)})",
-                expression_names={"#name": "name"},
-                expression_values=json.loads(dynamodb_json.dumps(expression_attribute_values)),
-            ).records
-        else:
-            json_response = self._scan().records
+        json_response = self._scan(
+            filter_expression=" AND ".join(filter_expressions) if filter_expressions else None,
```

Comment:
Do we still need this AND here?

Reply:
Yes - if there are multiple filter expressions they need to be joined with `AND` in this particular case. Options are:

- No filter expression in which case this is `None`
- Suspension only filter expression `suspended = :suspended`
- Specific project filter expression `#name IN (<list of names>)`
- Suspension and specific project filter expression `suspended = :suspended AND #name IN (<list of names>)`

Using `" AND ".join(filter_expressions)` lets us handle all of those cases.